syntax = "proto3";

package silenteight.datasource.api.nationalid.v1;

import "google/api/annotations.proto";

option java_package = "com.silenteight.datasource.api.nationalid.v1";
option java_outer_classname = "NationalIdProto";
option java_multiple_files = true;

service NationalIdInputService {
  rpc BatchGetMatchNationalIdInputs (BatchGetMatchNationalIdInputsRequest) returns (stream BatchGetMatchNationalIdInputsResponse) {
    option (google.api.http) = {
      post: "/agent-inputs/national-id/versions/v1/matches:batchGet"
      body: "*"
    };
  }
}

message BatchGetMatchNationalIdInputsRequest {
  // The list of matches data is requested for, in form of `alerts/<Alert ID>/matches/<Match ID>`.
  repeated string matches = 1;
  // The list of features the input data for each match is requested, in form of `features/<Feature ID>`.
  repeated string features = 2;
}

message BatchGetMatchNationalIdInputsResponse {
  repeated NationalIdInput national_id_inputs = 1;
}

message NationalIdInput {
  string match = 1;
  repeated NationalIdFeatureInput national_id_feature_inputs = 2;
}

message NationalIdFeatureInput {
  string feature = 1;

  // original document numbers
  repeated string alerted_party_document_numbers = 2;
  repeated string watchlist_document_numbers = 3;

  // countries from other sources
  repeated string alerted_party_countries = 4;
  repeated string watchlist_countries = 5;
}
